src/main/webapp/app/exercises/programming/shared/code-editor/service/code-editor-repository.service.ts

```diff
@@ -168,34 +168,37 @@ export class CodeEditorRepositoryFileService extends DomainDependentEndpointServ
         this.fileUpdateUrl = `${this.restResourceUrl}/files`;
     }
 
-    getRepositoryContent = () => {
-        return this.http.get<{ [fileName: string]: FileType }>(`${this.restResourceUrl}/files`).pipe(handleErrorResponse<{ [fileName: string]: FileType }>(this.conflictService));
+    getRepositoryContent = (domain?: DomainChange) => {
+        const restResourceUrl = domain ? this.calculateRestResourceURL(domain) : this.restResourceUrl;
+        return this.http.get<{ [fileName: string]: FileType }>(`${restResourceUrl}/files`).pipe(handleErrorResponse<{ [fileName: string]: FileType }>(this.conflictService));
     };
 
     /**
      * Gets the files of the repository and checks whether they were changed during a student participation.
      */
-    getFilesWithInformationAboutChange = () => {
-        return this.http
-            .get<{ [fileName: string]: boolean }>(`${this.restResourceUrl}/files-change`)
-            .pipe(handleErrorResponse<{ [fileName: string]: boolean }>(this.conflictService));
+    getFilesWithInformationAboutChange = (domain?: DomainChange) => {
+        const restResourceUrl = domain ? this.calculateRestResourceURL(domain) : this.restResourceUrl;
```

Comment:
As far as I can see this will not set this.restResourceUrl, so this variable might have a wrong value in future calls, doesn't it?

Reply:
This is on purpose! If you want to change the saved variable, you need to use the `setDomain` method. Passing arguments to the call directly are meant to only be used in context of this one call.

Reply:
I think you could be more explicit here and call it ``domainOverride`` or something so it's more obvious that this is overriding the currently stored rest resource URL